compiler/src/dotty/tools/dotc/transform/AccessProxies.scala

```diff
@@ -49,34 +47,60 @@ abstract class AccessProxies {
   /** Add all needed accessors to the `body` of class `cls` */
   def addAccessorDefs(cls: Symbol, body: List[Tree])(implicit ctx: Context): List[Tree] = {
     val accDefs = accessorDefs(cls)
+    transforms.println(i"add accessors for $cls: $accDefs%, %")
     if (accDefs.isEmpty) body else body ++ accDefs
   }
 
   trait Insert {
     import ast.tpd._
 
+    def accessorNameKind: ClassifiedNameKind
     def needsAccessor(sym: Symbol)(implicit ctx: Context): Boolean
 
     /** A fresh accessor symbol */
-    def newAccessorSymbol(owner: Symbol, name: TermName, info: Type, pos: Position)(implicit ctx: Context): TermSymbol = {
+    private def newAccessorSymbol(owner: Symbol, name: TermName, info: Type, pos: Position)(implicit ctx: Context): TermSymbol = {
       val sym = ctx.newSymbol(owner, name, Synthetic | Method, info, coord = pos).entered
       if (sym.allOverriddenSymbols.exists(!_.is(Deferred))) sym.setFlag(Override)
       sym
     }
 
+    /** An accessor symbol, create a fresh one unless one exists already */
+    private def accessorSymbol(owner: Symbol, accessorName: TermName, accessorInfo: Type, accessed: Symbol)(implicit ctx: Context) = {
+      def refersToAccessed(sym: Symbol) = accessedBy.get(sym) == Some(accessed)
+      owner.info.decl(accessorName).suchThat(refersToAccessed).symbol.orElse {
+        val acc = newAccessorSymbol(owner, accessorName, accessorInfo, accessed.pos)
+        accessedBy(acc) = accessed
+        acc
+      }
+    }
+
+    /** Rewire reference to refer to `accessor` symbol */
+    private def rewire(reference: RefTree, accessor: Symbol)(implicit ctx: Context): Tree = {
+      reference match {
+        case Select(qual, _) => qual.select(accessor)
+        case Ident(name) => ref(accessor)
```

Comment:
Do you need to desugar the Identifier here like in `tpd.becomes`?

Reply:
No, since we always to refer to given `accessor`.